zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLGenerator.scala

```diff
@@ -21,6 +21,19 @@ class AQLGenerator extends IQLGenerator {
     }
   }
 
+  def generate(query: IQuery, schemaMap: Map[String,Schema]): String = {
+    query match {
+      case q: Query =>
+        validateQuery(q)
+        parseQuery(q, schemaMap)
+      case q: CreateView => parseCreate(q, schemaMap(query.dataset))
+      case q: AppendView => parseAppend(q, schemaMap(query.dataset))
+      case q: UpsertRecord => parseUpsert(q, schemaMap(query.dataset))
```

Comment:
Oh, it's supposed to be used when we need to update the view's metadata. we can make it clear when @haochen07  clean up the dataset register work.